packages/daemon/src/service/trace-manager.ts

```diff
@@ -0,0 +1,250 @@
+import { Transform, TransformCallback } from 'stream';
+import { EventEmitter } from 'events';
+import { open, close, write } from 'fs-extra';
+import { provide, scope, ScopeEnum } from 'midway';
+import { StringDecoder } from 'string_decoder';
+import { generateId, PipelineStatus, PluginTypeI } from '@pipcook/pipcook-core';
+import Debug from 'debug';
+
+const debug = Debug('tracer');
+
+export type PipcookEventType = 'log' | 'job_status';
+
+/**
+ * base pipcook event, defined the fields `type` and `data`
+ */
+export class BasePipcookEvent {
```

Comment:
```suggestion
export class TraceEvent {
```

Reply:
Maybe interface?

Reply:
Interface will make us to define `type` in every child class.

> Moreover, we need to add unit tests for `Trace`, too.

Tests has been added for tracer and trace manager.